tests/ssl-opt.sh

```diff
@@ -9803,6 +9803,51 @@ run_test    "export keys functionality" \
             -c "EAP-TLS IV is:" \
             -s "EAP-TLS IV is:"
 
+# Test fragmentation of the client handshake
+requires_config_enabled MBEDTLS_SSL_HS_DEFRAG_CLT
+requires_config_enabled MBEDTLS_MPS_READER_C
+run_test    "TLS reassembly: fragmentation disabled (openssl server)" \
+            "$O_SRV -mtu 2048" \
+            "$P_CLI debug_level=3" \
+            0 \
+            -c "handshake: done"
+
+requires_config_enabled MBEDTLS_SSL_HS_DEFRAG_CLT
+requires_config_enabled MBEDTLS_MPS_READER_C
+run_test    "TLS reassembly: no fragmentation (openssl server)" \
+            "$O_SRV -mtu 512" \
+            "$P_CLI debug_level=3 max_frag_len=512" \
+            1 \
+            -c "TLS handshake fragmentation not enabled"
+
+requires_config_enabled MBEDTLS_SSL_HS_DEFRAG_CLT
+requires_config_enabled MBEDTLS_MPS_READER_C
+run_test    "TLS reassembly: some fragmentation (openssl server)" \
+            "$O_SRV -mtu 512" \
+            "$P_CLI debug_level=3 max_frag_len=512 max_hs_msg_len=2048" \
+            0 \
+            -c "handshake: done" \
+            -C "TLS handshake fragmentation not enabled"
```

Comment:
We should grep for debug output affirmative of the fragmentation. I'd suggest to dump a line `found fragmented TLS handshake message` analogous to the `found fragmented DTLS handshake message` that we use in the case of DTLS.